cmd/juju/charmhub/download.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charmhub
+
+import (
+	"context"
+	"net/url"
+	"os"
+	"syscall"
+
+	"github.com/juju/charm/v8"
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+
+	"github.com/juju/juju/api/modelconfig"
+	"github.com/juju/juju/charmhub"
+	"github.com/juju/juju/charmhub/transport"
+	jujucmd "github.com/juju/juju/cmd"
+	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/environs/config"
+)
+
+const (
+	downloadSummary = "Locates and then downloads a CharmHub charm."
+	downloadDoc     = `
+Download a charm from the CharmHub store by a specified name.
+
+Examples:
+    juju download postgresql
+
+See also:
+	info
+	find
+`
+)
+
+// NewDownloadCommand wraps downloadCommand with sane model settings.
+func NewDownloadCommand() cmd.Command {
+	return modelcmd.Wrap(&downloadCommand{})
+}
+
+// downloadCommand supplies the "download" CLI command used for downloading
+// charm snaps.
+type downloadCommand struct {
+	modelcmd.ModelCommandBase
+	out cmd.Output
+
+	modelConfigAPI ModelConfigGetter
+	charmHubClient CharmHubClient
+
+	charmHubURL   string
+	charmOrBundle string
+	archivePath   string
+}
+
+// Info returns help related download about the command, it implements
+// part of the cmd.Command interface.
+func (c *downloadCommand) Info() *cmd.Info {
+	download := &cmd.Info{
+		Name:    "download",
+		Args:    "[options] <charm>",
+		Purpose: downloadSummary,
+		Doc:     downloadDoc,
+	}
+	return jujucmd.Info(download)
+}
+
+// SetFlags defines flags which can be used with the download command.
+// It implements part of the cmd.Command interface.
+func (c *downloadCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.ModelCommandBase.SetFlags(f)
+	f.StringVar(&c.charmHubURL, "charm-hub-url", "", "specify url for querying the charmhub store")
+	f.StringVar(&c.archivePath, "path", "", "path to download the charm to")
```

Comment:
Suggestion filename instead of path.  Makes it more consistent with other commands. export-bundle.  

I'm not sure if a filename is required either.  Does path necessarily include a file?

Reply:
filepath is a lot better imo.